internal/pkg/coordinator/monitor.go

```diff
@@ -393,21 +396,26 @@ func (m *monitorT) getIPs() ([]string, error) {
 }
 
 func (m *monitorT) rescheduleUnenroller(ctx context.Context, pt *policyT, p *model.Policy) {
-	l := m.log.With().Str(dl.FieldPolicyId, pt.id).Logger()
+	u := uuid.Must(uuid.NewV4())
+	l := m.log.With().Str(dl.FieldPolicyId, pt.id).Str("unenroller_uuid", u.String()).Logger()
 	unenrollTimeout := time.Duration(p.UnenrollTimeout) * time.Second
 	if unenrollTimeout != pt.unenrollTimeout {
 		// unenroll timeout changed
-		if pt.unenrollCanceller != nil {
```

Comment:
The issue was that this was evauating to nil under one of the initial sets.
I think it's because `ensureLeadership` does not check if a key is valid which may lead to `rescheduleUnenroller` being called with a different `policyT` then is used in `handlePolicies`.
Tracking the canceller functions in a separate map resolves this issue

Reply:
this makes sense to me, but i'm curious if there's a functional reason to have more than one monitor running? would it make more sense to send the `policyT`s in through a channel to handle mutex?